### PR TITLE
Feature/basic functionalities

### PR DESCRIPTION
**Pull request by ermiyas111**

### CODE REVIEW

File: src/Services/Transaction.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private final ApiClient apiClient;
+    private final ObjectMapper objectMapper;
+
+    public Transaction(ApiClient apiClient) {
+        this.objectMapper = new ObjectMapper();
+        this.apiClient = apiClient;
+    }
+
+    public TransactionList getTransactionListByUser() throws IOException, InterruptedException, ExecutionException, NoSuchAlgorithmException, InvalidKeyException {
+        HttpResponse<String> response = apiClient.apiRequest("GET", "/transaction/find-by-user", "?p=1", null);
```
Comment (agebeyaw) [inline src/Services/Transaction.java]: what about getting the page number as param, defaulting to 1 if not specified?

File: src/Services/Transaction.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package Services;
```
Comment (agebeyaw) [inline src/Services/Transaction.java]: is it possible to have a root namespace appended, like `YaYa/Sdk/` or `YaYaSdk` so the package be like `YaYa/Sdk/Service` or `YaYaSdk/Service`?

File: src/ApiRequest/ApiClient.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+import Models.Time;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class ApiClient {
+
+    private static final String API_URL = System.getenv("API_URL");
```
Comment (agebeyaw) [inline src/ApiRequest/ApiClient.java]: May be some integrations use this SDK along with other SDKs in their system and better to name the env variables explicitly referencing YaYa, so the env variable names be better like `YAYA_API_URL`, `YAYA_API_KEY`, etc

**Review by agebeyaw — CHANGES_REQUESTED** [review]

File: src/Main.java
Comment (agebeyaw) [inline src/Main.java:1]: do we need this class if it just prints hello world?

File: src/com/yaya/sdk/Services/Transfer.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    private final ApiClient apiClient;
+    private final ObjectMapper objectMapper;
+
+    public Transfer(ApiClient apiClient) {
+        this.objectMapper = new ObjectMapper();
+        this.apiClient = apiClient;
+    }
+
+    public com.yaya.sdk.Models.Transfer[] getTransferList() throws IOException, InterruptedException, ExecutionException, NoSuchAlgorithmException, InvalidKeyException {
+        HttpResponse<String> response = apiClient.apiRequest("GET", "/transfer/", "", null);
```
Comment (agebeyaw) [inline src/com/yaya/sdk/Services/Transfer.java]: you can alternatively use `transfer/list` if you want to avoid the trailing slash

**Review by agebeyaw — COMMENTED** [review]
Airtime service seem not yet implemented, can you add that as well?

**Comment by ermiyas111** [comment]
> Airtime service seem not yet implemented, can you add that as well?

Buying airtime is not working

File: README.md [hunk 5]
```diff
@@ -1,2 +1,2 @@
-# yaya-python
+# yaya-python
```
Comment (agebeyaw) [inline README.md]: `yaya-java` ?

**Review by agebeyaw — APPROVED** [review]

**Pull request merged**